File: seed/pydantic-v2/trace/resources/v_2/resources/v_3/resources/problem/types/lightweight_problem_info_v_2.py
from pydantic import BaseModel
from typing import Set
from resources.commons.types.variable_type import VariableType


class LightweightProblemInfoV2(BaseModel):
    problem_id: str = Field(alias="problemId")
    problem_name: str = Field(alias="problemName")
    problem_version: int = Field(alias="problemVersion")
    variable_types: Set[VariableType] = Field(alias="variableTypes")

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/problem/types/lightweight_problem_info_v_2.rb
# frozen_string_literal: true

module Seed
  module V2
    module Problem
      module Types
        class LightweightProblemInfoV2 < Internal::Types::Model
          field :problem_id, -> { String }, optional: false, nullable: false
          field :problem_name, -> { String }, optional: false, nullable: false
          field :problem_version, -> { Integer }, optional: false, nullable: false
          field :variable_types, lambda {
            Internal::Types::Array[Seed::Commons::Types::VariableType]
          }, optional: false, nullable: false
        end
      end
    end
  end
end
